### ISSUE TITLE
TestStore does not expose state

### ISSUE THREAD

**Issue by lukeredpath** [pointfreeco/swift-composable-architecture#457]
I was wondering if it was a deliberate decision to not expose read-only access to the underlying state from `TestStore`.

Whilst generally you interact with state when asserting on state changes in send/receive calls, I think there are sometimes use cases for being able to write custom XCTest assertions based on the state and it feels like it would be useful to access it using `store.state`, especially with the new imperative API.

**Comment by stephencelis** [comment]
We're definitely open to changing this. At the moment we feel that assertions are the correct place to access state, but if you have some motivating examples we'd love to see them and consider them!

**Comment by lukeredpath** [comment]
So I've slept on this and given it a bit more thought, and I've changed my mind and now agree with you. 🤣

A bit of background - we're using what is essentially a ReSwift-compatible fork of TCA due to some legacy dependencies we can't drop overnight, but we hope to migrate to TCA in the future. As such I've been making a big effort to keep our library as API-compatible with TCA as possible.

After you dropped all of those new testing goodies the other day, I revisited our `TestStore` implementation which was largely written from scratch after I had trouble adapting the TCA version to our library. Things have progressed since then and I was able to drop the latest TCA `TestStore` into our project with only a few minor changes, which is great. However we did have some tests that were using `store.state` that my own implementation provided so I had to add it back to the updated implementation.

Upon reviewing the places we were doing this I realised that in every case there was a better alternative to reading the state in this way so I have managed to eliminate this and get rid of this entirely.
